lib/ansible/modules/template.py

```diff
@@ -79,6 +79,17 @@
     group: wheel
     mode: u=rw,g=r,o=r
 
+- name: Template a file using different marker strings
+  ansible.builtin.template:
+    src: globals.yml.j2
+    dest: /etc/kolla/globals.yml
+    block_start_string: "<%"
+    block_end_string: "%>"
+    variable_start_string: "<<"
+    variable_end_string: ">>"
+  vars:
+    vip_address: "<< hostvars['controller-00']['ansible_host'] >>"
```

Comment:
this is NOT limited to vars defined in task, but to ANY vars used in the template, no matter where they are defined.